.github/workflows/release.yml

```diff
@@ -0,0 +1,35 @@
+name: release
+
+on:
+  push:
+    tags:
+      - 'v[0-9]+.[0-9]+.[0-9]+*'
+
+permissions:
+  # Allow creating GitHub release
+  contents: write
+  # Allow closing associated milestone
+  issues: write
+
+jobs:
+  goreleaser:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          # Required for release notes
+          fetch-depth: 0
+      - id: go-version
+        # Reference: https://github.com/actions/setup-go/issues/23
+        run: echo "::set-output name=version::$(cat ./.go-version)"
+      - uses: actions/setup-go@v2
+        with:
+          go-version: ${{ steps.go-version.outputs.version }}
+      - name: Generate Release Notes
+        # Fetch CHANGELOG.md contents up to Git tag prior to this release, skipping top two lines
+        run: sed -n -e "1{/# /d;}" -e "2{/^$/d;}" -e "/# $(git describe --abbrev=0 --exclude="$(git describe --abbrev=0 --match='v*.*.*' --tags)" --match='v*.*.*' --tags | tr -d v)/q;p" CHANGELOG.md > /tmp/release-notes.txt
```

Comment:
This line is a bit difficult to grok at first glance; breaking it into a helper script may be advantageous. Your commenting is really well done, however, and helps greatly with readability!

Reply:
Ugh, this line. It's in a lot of places at the moment, which was why the one-liner was preferred over passing a script around to dozens of repositories. 😅  If we switch to something like [`changie`](https://changie.dev), I'm hoping we'll just change this over to `cat .changelog/${GITHUB_REF#refs/tags/}.md` and good riddance!